src/python/pants/jvm/resolve/coursier_setup.py

```diff
@@ -44,10 +44,32 @@
     classpath = dict()
     for dep in report['dependencies']:
         if not dep.get('file'):
-            raise Exception(
-                f"No jar found for {dep['coord']}. Check that it's available in the"
-                + " repositories configured in [coursier].repos in pants.toml."
-            )
+            if ":jar:sources:" in dep['coord']:
+                path = os.path.join(os.getcwd(), sys.argv[1] + ".empty")
+                f = open(path, "w")
+                f.write("")
+                f.close()
+                source = PurePath(path)
+                dest_name = dep['coord'].replace(":", "_")
+                classpath_dest = f"classpath/{dest_name}{ext}"
+                existing_source = classpath.get(classpath_dest)
+                if existing_source:
+                    if existing_source == source:
+                    # We've already captured this file.
+                        continue
+                    raise Exception(
+                        f"Duplicate jar name {classpath_dest} with incompatible source:\\n"
+                        f"  {source}\\n"
+                        f"  {existing_source}\\n"
+                    )
+                classpath[classpath_dest] = source
+                copyfile(source, classpath_dest)
+                continue
+            else:
```

Comment:
This is literally just a copy of the normal action but just passing an empty file for missing source jars (super-hacky yes). I noticed things upstream expect results to match 1-to-1 and in order, I just did this hack instead of finding every place where that was assumed. Since source jars aren't always a thing they'd have to be allowed to be not found

Reply:
For my education, do you have an example of how this setup script fails without this change?